services/main-frontend/src/components/page-specific/manage/courses/id/course-instances/ModuleCompletionReprocessButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { useTranslation } from "react-i18next"
+
+import { postReprocessModuleCompletions } from "../../../../../../services/backend/course-instances"
+import Button from "../../../../../../shared-module/components/Button"
+import OnlyRenderIfPermissions from "../../../../../../shared-module/components/OnlyRenderIfPermissions"
+import useToastMutation from "../../../../../../shared-module/hooks/useToastMutation"
+
+interface ModuleCompletionReprocessButtonProps {
+  courseInstanceId: string
+}
+
+const ModuleCompletionReprocessButton: React.FC<ModuleCompletionReprocessButtonProps> = ({
+  courseInstanceId,
+}) => {
+  const { t } = useTranslation()
+  const postReprocessCompletionsMutation = useToastMutation(
+    async () => {
+      return postReprocessModuleCompletions(courseInstanceId)
+    },
+    { notify: true, method: "POST" },
+    // { onError: setMutationError },
+  )
+
+  return (
+    <OnlyRenderIfPermissions action={{ type: "edit" }} resource={{ type: "global_permissions" }}>
+      <Button
+        variant="primary"
+        size="medium"
+        onClick={() => postReprocessCompletionsMutation.mutate()}
```

Comment:
Prolly should add a confirm like here: https://github.com/rage/secret-project-331/blob/1168f9a51e98c8726593cf16fa6adae30e41c106/services/main-frontend/src/components/page-specific/manage/courses/id/pages/ManageCourseStructure.tsx#L139-L146.

Otherwise someone might trigger this accidentally.

Reply:
Makes sense, I will add it.